clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1079,16 +1028,16 @@ protected void produceToStoreBufferServiceOrKafka(
       int kafkaClusterId) throws InterruptedException {
     long totalBytesRead = 0;
     double elapsedTimeForPuttingIntoQueue = 0;
-    int subPartition = PartitionUtils.getSubPartition(topicPartition, amplificationFactor);
     boolean metricsEnabled = emitMetrics.get();
     long beforeProcessingBatchRecordsTimestampMs = System.currentTimeMillis();
     for (PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> record: records) {
       long beforeProcessingPerRecordTimestampNs = System.nanoTime();
-      PartitionConsumptionState partitionConsumptionState = partitionConsumptionStateMap.get(subPartition);
+      PartitionConsumptionState partitionConsumptionState =
+          partitionConsumptionStateMap.get(topicPartition.getPartitionNumber());
       if (partitionConsumptionState != null) {
         partitionConsumptionState.setLatestPolledMessageTimestampInMs(beforeProcessingBatchRecordsTimestampMs);
       }
-      if (!shouldProcessRecord(record, subPartition)) {
+      if (!shouldProcessRecord(record, topicPartition.getPartitionNumber())) {
```

Comment:
we can remove partition arg from `shouldProcessRecord` since record already contains partition number: `record.getPartition()`